hazelcast/src/main/java/com/hazelcast/map/impl/PartitionContainer.java

```diff
@@ -239,6 +239,26 @@ public void setLastCleanupTimeCopy(long lastCleanupTimeCopy) {
         this.lastCleanupTimeCopy = lastCleanupTimeCopy;
     }
 
+    /**
+     * Cleans up the container's state if the enclosing partition is migrated
+     * off this member. Whether cleanup is needed is decided based on the
+     * provided {@code replicaIndex}.
+     *
+     * @param replicaIndex The replica index to use for deciding per map whether
+     *                     cleanup is necessary or not
+     */
+    final void cleanUpOnMigration(int replicaIndex) {
+        mapService.getMapServiceContext().getMapContainers().keySet()
+                .stream()
+                .filter(mapName -> replicaIndex == -1
+                        || replicaIndex > getRecordStore(mapName).getMapContainer().getTotalBackupCount())
```

Comment:
minor:
>replicaIndex > getRecordStore(mapName).getMapContainer().getTotalBackupCount())

Not to replicate same logic, we have `lesserBackupMapsThen` static method in `MapMigrationAwareService`, maybe it can also be used here.

Reply:
Addressed in https://github.com/hazelcast/hazelcast/pull/21517/commits/708f0f5bed9d7a3b265c7b6d9ae91b7b506033e6.